git/operations.go

```diff
@@ -137,12 +137,26 @@ func revlist(path, ref string) ([]string, error) {
 }
 
 // Return the revisions and one-line log commit messages
-func onelinelog(path, refspec string) ([]Commit, error) {
+func onelinelog(path, refspec, gitpath string) ([]Commit, error) {
 	out := &bytes.Buffer{}
+
+	// we need to distinguish whether gitpath is populated or not,
+	// because supplying an empty string to execGitCmd results in git complaining about
+	// >> ambiguous argument '' <<
+	if gitpath != "" {
+		if err := execGitCmd(path, nil, out, "log", "--oneline", "--no-abbrev-commit", refspec, gitpath); err != nil {
+			fmt.Printf("\n onelinelog ERROR: %v\n\n", err)
+			return nil, err
+		}
+		return splitLog(out.String())
+	}
+
 	if err := execGitCmd(path, nil, out, "log", "--oneline", "--no-abbrev-commit", refspec); err != nil {
+		fmt.Printf("\n onelinelog ERROR: %v\n\n", err)
```

Comment:
Debug printf?

Reply:
Sorry about the debug printouts. I thought I cleaned up - obviously now.